charity-data/getcharities.py

```diff
@@ -0,0 +1,80 @@
+from urllib import parse
+import requests
+import json
+
+base_url = 'https://api.data.charitynavigator.org/v2/Organizations?'
+
+# Parameters for request URL
+app_id = '80de6d65'
+app_key = '269363d5f87d1f1834f7b5ef5f9e4c3a'
+pageSize = '1000'
+rated = 'true'
+sizeRange = '3'
+sort = 'RATING'
+params = {'app_id': app_id, 'app_key': app_key, 'pageSize': pageSize,
+          'rated': rated, 'sizeRange': sizeRange, 'sort': sort}
+
+url = base_url + parse.urlencode(params)
+
+# Get JSON response body
+charity_nav_json = requests.get(url).json()
+
+# Counters for the number of charities with no valid donate link, the number
```

Comment:
This code could be split up into functions.

Reply:
+1

Reply:
I created a helper function for getting the charity's link. I considered making one for getting the image link, but that's only an if statement, so modularization of that seemed excessive.

Reply:
I think what's intended here is that you code should be modularized into testable and reusable blocks and not a bunch of statement at the top level. Please break the logic into  independent functions. eg. get_charities_json(), process_charity(charity), getCharityLink(charity), getCharityImage(charity)

Also please use the if __name__ == "__main__" construct in your scripts. https://www.freecodecamp.org/news/if-name-main-python-example/#:~:text=We%20can%20use%20an%20if,name%20if%20it%20is%20imported.

Reply:
Done.